impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -407,6 +432,11 @@ public synchronized void start() {
                     clusterConfig.getAnonConnectConfigForCluster(buildBrokerListFor(Listener.ANON)),
                     READY_TIMEOUT_SECONDS, TimeUnit.SECONDS,
                     clusterConfig.getBrokersNum());
+            if (clusterConfig.isSaslScram() && !clusterConfig.getUsers().isEmpty()) {
+                if (clusterConfig.isKafkaPostVersion41()) {
```

Comment:
Could we switch to this approach wholesale (i mean <4.0)?   Would it be much work. The motivation to to have fewer conditions.